hydra-cluster/src/Hydra/Cluster/Faucet.hs

```diff
@@ -126,6 +109,58 @@ seedFromFaucet_ ::
 seedFromFaucet_ node vk ll marked tracer =
   void $ seedFromFaucet node vk ll marked tracer
 
+-- | Return the remaining funds to the faucet
+returnFundsToFaucet ::
+  Tracer IO FaucetLog ->
+  RunningNode ->
+  Actor ->
+  IO ()
+returnFundsToFaucet tracer RunningNode{networkId, nodeSocket} sender = do
+  (faucetVk, _) <- keysFor Faucet
+  let faucetAddress = mkVkAddress networkId faucetVk
+
+  (senderVk, senderSk) <- keysFor sender
+  utxo <- queryUTxOFor networkId nodeSocket QueryTip senderVk
+
+  retryOnExceptions tracer $ do
+    let allLovelace = selectLovelace $ balance @Tx utxo
+    -- XXX: Using a hard-coded high-enough value to satisfy the min utxo value.
+    -- NOTE: We use the faucet address as the change deliberately here.
+    tx <- sign senderSk <$> buildTxBody utxo faucetAddress 1_000_000
```

Comment:
is this the only way we can calculate the fee of the network?

Reply:
I guess this is one of the few possible ways...there are functions in cardano api to evaluate fees but here we already have access to `buildTransaction` so we build it twice so that the first time around we can inspect the fees. What did you have in mind?

Reply:
I was thinking about the cardano-api but it is fine to use this instead as you already have it in hand.

Anyhow, I would create a helper function with a proper name, as it is not trivial to tell the intention of this line unless u see its usage downstream.